src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -2083,24 +2120,17 @@ private bool IsTypeDeclarationStart()
                 case SyntaxKind.StructKeyword:
                     return true;
 
+                case SyntaxKind.ExplicitKeyword or SyntaxKind.ImplicitKeyword
+                    when this.PeekToken(1).ContextualKind is SyntaxKind.ExtensionKeyword:
+
+                    return IsCompatBreakingFeatureEnabled(MessageID.IDS_FeatureExtensions);
```

Comment:
how is there a compat break here?  this feels like `yield return`.  Because of implicit/explicit, we always know this is an extension and can safely parse it out.

Reply:
Yup, implicit/explicit are now required.